docs/flow/concepts/components.md

```diff
@@ -6,3 +6,111 @@ weight: 100
 ---
 
 # Components
+
+_Components_ are the building blocks of Grafana Agent Flow. Each component is
+responsible for handling a single task, such as retrieving secrets or
+collecting Prometheus metrics.
+
+Components are composed of two parts:
+
+* Arguments: settings which configure a component
+* Exports: named values which a component exposes to other components
+
+Each component has a name which describes what that component is responsible
+for. For example, the `local.file` component is responsible for retrieving the
+contents of files on disk.
+
+Components are specified by users by first providing the component's name with
+a user-specified label, and then by providing arguments to configure the
+component.
+
+> Components are referenced by combining the component name with its label. For
+> example, a `local.file` component labeled `foo` would be referenced as
+> `local.file.foo`.
+>
+> The combination of a component's name and its label must be unique within the
+> configuration file.
+
+## Pipelines
+
+Most arguments a user specifies for a component will be some constant value,
+such setting a `log_level` attribute to the quoted string `"debug"` (`log_level
+= "debug"`).
+
+Users may also use _expressions_ to dynamically compute the value of an
+argument at runtime. Among other things, expressions may be used to retrieving
+the value of an environment variable (`log_level = env("LOG_LEVEL")`) or
+referencing the export of another component (`log_level =
+local.file.log_level.contents`).
+
+When a user configures a component's argument to reference an export of another
+component, a relationship is created: a component's input (arguments) now
+depends on another component's output (exports). The input of the component
+will now be re-evaluated any time the exports of the components it references
+get updated.
+
+The flow of data through the set of references between components forms a
+_pipeline_.
+
+An example pipeline may look like this:
+
+1. A `local.file` component watches a file on disk containing an API key.
```

Comment:
Doesn't really matter as Markdown will render the numbered list correctly, but the numbering is off.